Exceptions/Code is not compliant with C++17

C++17 removed dynamic exception specifications / throw. Before that, they had been deprecated since C++11. 

GCC 7.3.0 in C++17 Mode reports these errors:

plusaes.hpp:197:57: error: ISO C++1z does not allow dynamic exception specifications
 inline unsigned int get_round_count(const int key_size) throw(std::invalid_argument) {
                                                         
plusaes.hpp:210:75: error: ISO C++1z does not allow dynamic exception specifications
 inline RoundKeys expand_key(const unsigned char *key, const int key_size) throw(std::invalid_argument) {

--- a/include/plusaes/plusaes.hpp
+++ b/include/plusaes/plusaes.hpp
@@ -194,7 +194,11 @@ inline Word rot_word(const Word v) {
     return ((v >> 8) & 0x00FFFFFF) | ((v & 0xFF) << 24);
 }
 
-inline unsigned int get_round_count(const int key_size) throw(std::invalid_argument) {
+/**
+ * @private
+ * @throws std::invalid_argument
+ */
+inline unsigned int get_round_count(const int key_size) {
     switch (key_size) {
     case 16:
         return 10;
@@ -207,7 +211,11 @@ inline unsigned int get_round_count(const int key_size) throw(std::invalid_argum
     }
 }
 
-inline RoundKeys expand_key(const unsigned char *key, const int key_size) throw(std::invalid_argument) {
+/**
+ * @private
+ * @throws std::invalid_argument
+ */
+inline RoundKeys expand_key(const unsigned char *key, const int key_size) {
     if (key_size != 16 && key_size != 24 && key_size != 32) {
         throw std::invalid_argument("Invalid key size");
     }

